runtime/lua/vim/net.lua

```diff
@@ -0,0 +1,30 @@
+local M = {}
+
+function M.download(url, opts)
+  vim.validate({
+    url = { url, 'string' },
+    opts = { opts, 'table' },
+  })
+
+  local args = {
+    'curl',
+    '--no-progress-meter',
```

Comment:
maybe `--silent --fail --show-error` would be more appropriate here?

Reply:
Do we want to add a small `--retry` to handle flakes? (CI, I'm looking at you...)